src/hooks/useSpendingLimitGas.ts

```diff
@@ -0,0 +1,32 @@
+import { BigNumber } from 'ethers'
+import { useWeb3 } from '@/hooks/wallets/web3'
+import { getSpendingLimitContract } from '@/services/contracts/spendingLimitContracts'
+import useAsync from '@/hooks/useAsync'
+import { SpendingLimitTxParams } from '@/components/tx/modals/TokenTransferModal/ReviewSpendingLimitTx'
+import useChainId from '@/hooks/useChainId'
+
+const useSpendingLimitGas = (params: SpendingLimitTxParams) => {
+  const chainId = useChainId()
+  const provider = useWeb3()
+
+  const [gasLimit, error, loading] = useAsync<BigNumber>(() => {
+    if (!provider) return
+
+    const contract = getSpendingLimitContract(chainId, provider.getSigner())
+
+    return contract.estimateGas.executeAllowanceTransfer(
+      params.safeAddress,
+      params.token,
+      params.to,
+      params.amount,
+      params.paymentToken,
+      params.payment,
+      params.delegate,
+      params.signature,
+    )
+  }, [provider, chainId, params])
+
+  return { gasLimit, error, loading }
```

Comment:
Suggestion to rename error and loading to `gasLimitError` and `gasLimitLoading` to mimic the regular gas limit hook.